### ISSUE TITLE
snap helm cannot read configs

### ISSUE THREAD

**Issue by ryan-peachey-seon** [snapcrafters/helm#10]
There seems to be something weird with the snap version of helm as it's not properly reading cluster configs.  The default via environment config is ignored, and even the manually specified parameter is also ignored.

config location:
Specific: `~/.kube/dev-k8s`
Generic: `~/.kube/<env>-k8s`

Environment Variable set:
```
$ echo $KUBECONFIG
/home/rpeachey/.kube/dev-k8s
```
Installation:
```
$ sudo snap refresh helm --edge
helm (edge) 2.6.2 from 'snapcrafters' refreshed
$ which helm
/snap/bin/helm
```
but the helm app is completely broken without forcing a single config file in a single location?
```
$ helm version
Please copy Kubernetes config to /home/rpeachey/snap/helm/common/kube/config.
```
This completely ignores the env variable KUBECONFIG  and also completely ignores the --kubeconfig parameter introduced in what I believe was v 2.6.1
```
$ helm --kubeconfig ~/.kube/dev-k8s version
Please copy Kubernetes config to /home/rpeachey/snap/helm/common/kube/config.
```

When I manually install helm from tgz, this non-default configuration is not required
Installation:
```
$ mkdir helm && cd helm
$ wget https://storage.googleapis.com/kubernetes-helm/helm-v2.6.2-linux-amd64.tar.gz
$ tar -zxvf helm-v2.6.2-linux-amd64.tar.gz
$ cd ~ && mkdir bin
$ ln -sf ~/helm/linux-amd64/helm ~/bin/helm
```
helm just works with my config via environment variable
```
$ helm init
Creating /home/rpeachey/.helm 
Creating /home/rpeachey/.helm/repository 
Creating /home/rpeachey/.helm/repository/cache 
Creating /home/rpeachey/.helm/repository/local 
Creating /home/rpeachey/.helm/plugins 
Creating /home/rpeachey/.helm/starters 
Creating /home/rpeachey/.helm/cache/archive 
Creating /home/rpeachey/.helm/repository/repositories.yaml 
$HELM_HOME has been configured at /home/rpeachey/.helm.

Tiller (the Helm server-side component) has been installed into your Kubernetes Cluster.
Happy Helming!
```
```
$ helm version
Client: &version.Version{SemVer:"v2.6.2", GitCommit:"be3ae4ea91b2960be98c07e8f73754e67e87963c", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.6.2", GitCommit:"be3ae4ea91b2960be98c07e8f73754e67e87963c", GitTreeState:"clean"}
```

**Comment by joedborg** [comment]
Hi @ryan-peachey-seon,

To my knowledge of Snapd, even with the `home` plug, snapped apps cannot read hidden dirs and files.  A workaround would be to set `$KUBECONFIG` to a non-hidden directory.

**Comment by ryan-peachey-seon** [comment]
@joedborg That doesn't sound right.  kubectl is also installed through snap and it doesn't seem to have any issues accessing the specified config file(s).  That being said, I'm not 100% sure that I've only ever used the snap kubectl binary and not inadvertently used a different binary that happened to get installed outside of snap.  I'll keep an eye out going forward to see if I can confirm this or not.

**Comment by joedborg** [comment]
Hi @FredericLeroy, @ryan-peachey-seon,

I have opened https://forum.snapcraft.io/t/classic-confinement-for-existing-helm-snap/4375/1 to get the snap moved to classic confinement as I agree that it should be as `kubectl`.

**Comment by jcassee** [comment]
After installing using `--classic`, calling `helm` without the wrapper seems to work fine. No need for `~/snap/helm/common/kube/config`.

**Comment by matohavo** [comment]
`sudo snap install --classic helm` didn't help, does this need to be merged first?

**Comment by gjergjkastrioti** [comment]
`sudo snap install --classic helm`  didn't work. I gave up trying Snap and used the Script installation instead, works fine.

**Comment by PaulineLc** [comment]
The issue does not seem fixed indeed.

```
$ snap install helm --classic
helm 2.9.1 from 'snapcrafters' installed
$ helm 
Please copy Kubernetes config to /home/pauline/snap/helm/common/kube/config.
```

Downloading the binaries and placing them in your path works fine though.

**Comment by mhjort** [comment]
Same issue here

**Comment by Oblynx** [comment]
same here

**Comment by jpnarkinsky** [comment]
here as well.  Finally just copied the file in desperation, but this is a serious pain in the neck.

**Comment by dankirkpatrick** [comment]
This is still an issue in 2.9.1 Helm, on both Stable and Edge.  Why was this closed?

**Comment by ricardofbarros** [comment]
My current workaround for this:

```
sudo snap install helm --classic
ln -s /home/$(whoami)/.kube/config /home/$(whoami)/snap/helm/common/kube/config
```

**Comment by codycraven** [comment]
> This is still an issue in 2.9.1 Helm, on both Stable and Edge. Why was this closed?

This isn't closed. The reference above your comment is for another repository by jonashackt having to do with Ansible.

**Comment by joedborg** [comment]
Snap has been classic for a while, so closing.  If this is still an issue for anyone, please speak up and I'll reopen.

**Comment by clovis-dugue** [comment]
Hello @joedborg ! I just encountered this issue in a weird fashion today.  

With the latest (`version.BuildInfo{Version:"v3.5.4", GitCommit:"1b5edb69df3d3a08df77c9902dc17af864ff05d1", GitTreeState:"clean", GoVersion:"go1.15.11"}`) helm (and snap) version, helm still failed silently with error code 1 on snap install.  

However, the issue only occurred when I ran helm through the `child_process` typescript library.  

Uninstalling the snap, and using helm's version from the provided install script fixed the issue ; meaning the problem does originate from the snap.  

Below is the code that yielded problems using the snap:
```ts
const namespace = "default";
const release = "test";
const command = "helm";
const args = ["get", "values", "-n", namespace, release, "-o", "json"];
// NOTE: Manually adding the kube context in the parameters did not work since my kubeconfig is stored in the default ~/.kube/config location.
const output = child_process.spawnSync(command, args);

// DEBUG: These 2 members are empty
console.log(output.stdout);
console.log(output.stderr);

if (output.error !== undefined) {
    console.error(`${command} ${args.join(' ')} failed. Reason: ${output.error}`);
    throw output.error;
}
if (output.status !== 0) {
    console.error(`${command} ${args.join(' ')} failed with code: ${output.status}`);
    throw new Error(`code: ${output.status}`);
}

return output.stdout.toString();
```

I would like to emphasize, running the exact same command directyl from the terminal worked just fine. I did not try [Ricardo's workaround](https://github.com/snapcrafters/helm/issues/10#issuecomment-419645091) as I only stumbled upon this after changing my helm install.

I do not have any idea what the problem may be here. Please tell me if I should open a new issue (all of the relevant details should be here).
